Source/WebKit/UIProcess/API/glib/WebKitImage.cpp

```diff
@@ -0,0 +1,381 @@
+/*
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WebKitImage.h"
+
+#include "WebKitImagePrivate.h"
+
+#include <wtf/Assertions.h>
+#include <wtf/Hasher.h>
+#include <wtf/glib/GRefPtr.h>
+#include <wtf/glib/GUniquePtr.h>
+#include <wtf/glib/WTFGType.h>
+
+static void webkit_image_gicon_interface_init(GIconIface*);
+static void webkit_image_gloadable_icon_interface_init(GLoadableIconIface*);
+
+struct _WebKitImagePrivate {
+    int width;
+    int height;
+    unsigned stride;
+    CString type;
+    WebKitImageCreateBytesFunction createBytesFn;
+
+    GRefPtr<GBytes> bytes;
+};
+
```

Comment:
It would be nice to add here a paragraph or two explaining what the `WebKitImage` type is. We have such blurbs for most types, see for example [WebKitUserContentManager](https://wpewebkit.org/reference/stable/wpe-webkit-2.0/class.UserContentManager.html) or [WebKitBackForwardList](https://wpewebkit.org/reference/stable/wpe-webkit-2.0/class.BackForwardList.html), and I think it helps understand the API better. Plus, we can indicate in which WebKit version the class was added.


```suggestion
/**
 * WebKitImage:
 *
 * Does beep-meep-zorp and cures all imaginable diseases. It's magic!
 *
 * Since: 2.52
 */
```